VotingApplication/VotingApplication.Web/Api/Controllers/AccountController.cs

```diff
@@ -153,6 +174,64 @@ public IHttpActionResult Logout()
             return Ok();
         }
 
+        // POST api/Account/ForgotPassword
+        [AllowAnonymous]
+        [Route("ForgotPassword")]
+        public async Task<IHttpActionResult> ForgotPassword(ForgotPasswordBindingModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var user = await UserManager.FindByNameAsync(model.Email);
+
+            // Currently email confirmation is not set up
+            if (user == null /*|| !(await UserManager.IsEmailConfirmedAsync(user.Id))*/ )
+            {
+                ModelState.AddModelError("", "The user either does not exist or is not confirmed.");
+                return BadRequest(ModelState);
+            }
+
+            string code = await UserManager.GeneratePasswordResetTokenAsync(user.Id);
+
+            var callbackUrl = Url.Content("~/#/Account/ResetPassword?email=") + HttpUtility.UrlEncode(user.Email) + "&code=" + HttpUtility.UrlEncode(code);
+
+            await UserManager.SendEmailAsync(user.Id, "Pollster Password Reset", "Please reset your password by clicking <a href=\"" + callbackUrl + "\">here</a>");
```

Comment:
General rule of thumb, but you should avoid "click here" links: http://uxmovement.com/content/why-your-links-should-never-say-click-here/

I'll push this through and fix in a separate commit though.